core/shared/src/main/scala/fs2/text.scala

```diff
@@ -212,60 +212,19 @@ object text {
       (out, carry)
     }
 
-    def extractLines(
-        buffer: Vector[String],
-        chunk: Chunk[String],
-        pendingLineFeed: Boolean
-    ): (Chunk[String], Vector[String], Boolean) = {
-      @tailrec
-      def go(
-          remainingInput: Vector[String],
-          buffer: Vector[String],
-          output: Vector[String],
-          pendingLineFeed: Boolean
-      ): (Chunk[String], Vector[String], Boolean) =
-        if (remainingInput.isEmpty)
-          (Chunk.indexedSeq(output), buffer, pendingLineFeed)
-        else {
-          val next = remainingInput.head
-          if (pendingLineFeed)
-            if (next.headOption == Some('\n')) {
-              val out = (buffer.init :+ buffer.last.init).mkString
-              go(next.tail +: remainingInput.tail, Vector.empty, output :+ out, false)
-            } else
-              go(remainingInput, buffer, output, false)
-          else {
-            val (out, carry) = linesFromString(next)
-            val pendingLF =
-              if (carry.nonEmpty) carry.last == '\r' else pendingLineFeed
-            go(
-              remainingInput.tail,
-              if (out.isEmpty) buffer :+ carry else Vector(carry),
-              if (out.isEmpty) output
-              else output ++ ((buffer :+ out.head).mkString +: out.tail),
-              pendingLF
-            )
+    def go(stream: Stream[F, String], buffer: String, first: Boolean): Pull[F, String, Unit] =
+      stream.pull.uncons.flatMap {
+        case None => if (first) Pull.done else Pull.output1(buffer)
+        case Some((chunk, stream)) =>
+          val (b, vector) = chunk.foldLeft((buffer, Vector.empty[String])) {
```

Comment:
There may be an opportunity here for further optimization by replacing the foldLeft and incremental vector creation with a loop and mutable Buffer and then using Chunk.buffer.

Reply:
```
[info] LinesBenchmark.linesBenchmark                0            4  thrpt    3  79623,427 ±  5563,967  ops/s
[info] LinesBenchmark.linesBenchmark                0           16  thrpt    3  77606,758 ± 13907,551  ops/s
[info] LinesBenchmark.linesBenchmark                0           64  thrpt    3  77267,067 ± 15156,153  ops/s
[info] LinesBenchmark.linesBenchmark                1            4  thrpt    3  21230,719 ±   962,499  ops/s
[info] LinesBenchmark.linesBenchmark                1           16  thrpt    3  39957,504 ±   356,596  ops/s
[info] LinesBenchmark.linesBenchmark                1           64  thrpt    3  49001,153 ±  1920,321  ops/s
[info] LinesBenchmark.linesBenchmark               10            4  thrpt    3   5634,845 ±   237,259  ops/s
[info] LinesBenchmark.linesBenchmark               10           16  thrpt    3  15571,785 ±  4894,706  ops/s
[info] LinesBenchmark.linesBenchmark               10           64  thrpt    3  32902,864 ±  3014,822  ops/s
[info] LinesBenchmark.linesBenchmark              100            4  thrpt    3    637,273 ±    57,592  ops/s
[info] LinesBenchmark.linesBenchmark              100           16  thrpt    3   2150,364 ±    46,006  ops/s
[info] LinesBenchmark.linesBenchmark              100           64  thrpt    3   6350,043 ±  8177,242  ops/s
```